### PR TITLE
Rails 6!

### PR DESCRIPTION
**Pull request by gregorbg**
This is not the most recent version (we're at `6.0.4` when there's already `6.1.4`) but `ENVied` pins us to the `6.0` branch because of a transitive dependency on `thor`, see https://github.com/thewca/worldcubeassociation.org/issues/6330. Nonetheless, this establishes basic compatibility with Rails 6 and upgrading to `6.1` releases should be a breeze!

The way I went about upgrading the config files is as follows: I manually generated a new `dummy` application via `rails new dummy` and adding ReactJS via `bin/rails webpacker:install:react`. Then I went through each config file in the project and manually diff'ed the two versions, porting over the missing bits and pieces.

If certain configuration options are commented out, then (a) I decided they were not useful for our project or (b) Rails had commented them out already in the first place (this is especially the case with files that contain only commented-out lines, I checked them in anyways for clarity's sake)

It should be noted that Sprocket assets pipeline has been deprecated in favor of Webpacker **for JavaScript**. We should move everything from `app/assets/javacripts` to `app/webpacker/javascript`, but this is manual migration work and this change will occur anyways when slowly migrating parts of the website to React. So no need to specifically push this endeavour imho.

### CODE REVIEW

**Comment by gregorbg** [comment]
All the deprecations are fixed, unfortunately we're running into a bug in Rails 6: https://github.com/rails/rails/issues/36965

**Comment by saranshgrover** [comment]
No technical comments, but thanks for your work on this!! Great to see us upgrading important dependencies like this. 😀

**Comment by danieljames-dj** [comment]
Hi @gregorbg - thanks a lot for the work. This branch is working perfectly in my M1 Mac. The existing Rails version was crashing in M1.

File: WcaOnRails/app/models/user.rb [hunk 1]
```diff
@@ -238,6 +238,12 @@ def claim_wca_id_validations
     end
   end
 
+  # workaround / very nasty hotfix for https://github.com/rails/rails/issues/36965
```
Comment (jonatanklosko) [inline WcaOnRails/app/models/user.rb:241]: Just so it's easy to locate later :)

```suggestion
  # TODO: remove this workaround once https://github.com/rails/rails/issues/36965 is fixed
```

File: WcaOnRails/config/cable.yml [hunk 2]
```diff
@@ -6,4 +6,5 @@ test:
 
 production:
   adapter: redis
-  url: redis://localhost:6379/1
+  url: <%= ENV.fetch("REDIS_URL") { "redis://localhost:6379/1" } %>
+  channel_prefix: dummy_production
```
Comment (jonatanklosko) [inline WcaOnRails/config/cable.yml:10]: I guess it should be `wca_production`, but this config doesn't seem relevant anyway.

**Review by jonatanklosko — APPROVED** [review]
Great update!

**Pull request merged**